### PR TITLE
Update logits to new mlx + version bump

### PR DESCRIPTION
**Pull request by awni**
- Use the model dtype for logits and loss to save RAM (since logsumexp accumulates in high precision this should be ok). Only cast the losses (which are much smaller) prior to the reduction.
- Use `mx.fast.rms_norm` in plamo2

### CODE REVIEW

**Review by barronalex — APPROVED** [review]
Looks good!
